### ISSUE TITLE
emit a sound when North is reached

### ISSUE THREAD

**Issue by lucamp** [Kr0oked/Compass#8]
Hi,
could you please consider adding a sound when North is reached?
Since some users may not want it, it should be possible to enable and disable with a button near the actual ones

Speaking about where you can find open source sounds, I found:

[https://github.com/akx/Notifications](https://github.com/akx/Notifications)

where "Sharp.ogg" inside the OGG folder seems good enough, a very short sound not too intrusive

_v 1.3 from F-Droid_

Thanks

**Comment by Kr0oked** [comment]
Hi,

when do you think this feature comes in handy?
At least in my uses cases I dont want to align the phone to north.

**Comment by lucamp** [comment]
When I was in Morocco I found really difficult to have basic orientation inside a medina, it's the old center of a city, with higher building (not skyscrapers) and small alleys. Sometimes even locating the sun direction is difficult. For instance you can search photos of the Fez's medina (highly recommended to visit).
Nothing bad never happened to me but it not recommendable to always walk with a device in your hand, also look at it for just a moment and you will be immediately surrounded by touts that want to bring you where they want. Even if some of them are nice and polite, it can be pretty hard to get rid of the "not so good" ones.

Having something in your pocket that sounds (or even vibrates) when you are facing north can help you to locate immediately other directions, even if it will not be precise you can guess at least if you are heading completely in the wrong directions. Turning around 360° disguising (or looking for real) it like you are admiring the buildings or the shops does not drag to much attentions. Also a small beep is not an issue since there are plenty of other noises.

Another use can be roughly locating when looking for mushrooms in a dense forest.

In both cases smartphone gps signal can have issue and it will drain the battery really fast compared to just a compass

**Comment by Kr0oked** [comment]
A trip to Morocco would definetly be interessting. Hopefully the pandemic will pass by quickly and traveling will be easier again.
At the moment I think about implementing a feature that alllows you to set your "direction of march" and shows your current offset. So you face the compass to 90° and lock this as your direction of march. Maybe it could then vibrate when the device points in this direction.
But to be honest I think your use case should be implemented in another specialised app. This app could focus on navigation from within your pocket and could stay active even with your dispaly turned off. And the current app shows the direction in which the tip of the device points. But when you have your phone in your pocket you want to know in which direction your screen or the phones back points.

**Comment by OmlineEditor** [comment]
if you add this function, please note that in the sideways position in the pocket, it may be incorrect to determine the direction to the north. it is necessary to somehow read the position of the phone and check the north in accordance with the position

**Comment by hackerb9** [comment]
I notice this issue is marked "won't fix", but no reason is given. Does that mean an option to pick a specific direction as special would also be out of the question? 

Genuine compasses for orienteering can be rotated so the direction you want to go is up. Perhaps, this compass could do the same?

Or, would that also be "won't fix"? If so, please ignore this with my apologies.

**Comment by Kr0oked** [comment]
@hackerb9 I am currently not planning to implement the north specific feature. But I am still thinking about implementing a "direction of march" feature like displayed here in the app Trail sense. There it is possible to pin a location and then the app shows the current deviation.
![Screenshot_20240525-223014](https://github.com/Kr0oked/Compass/assets/5984532/73a0c670-2870-429e-8d76-2c1a7f74625a)

**Comment by hackerb9** [comment]
Yeah, that's pretty much what I'm hoping for. Trail Sense is extreme overkill, though. Something simpler (and more attractive) would be nice. 

You probably already have a good idea how you want it to work, but if not, perhaps something like this would be simple and not intrusive to people who don't use the feature:

1. If user holds down in center of compass, the current direction is locked in as the "direction of march". 
2. A yellow arrow appears on the rotating dial. The arrow turns green when on track.
3. Optionally,  emit a continuous tone changing in pitch so one can navigate without having look down.
   a. E.g., dead on is a 440Hz sine, going down in pitch for each degree off track. 
   b. Fanciness like 3D/stereo panning (so people intuitively turn towards the tone) would be a nice touch but not necessary.
